### PR TITLE
Update from.md

### PR DESCRIPTION
**Pull request by thomoco**
Updated the SELECT...FROM...FINAL documentation to add better guidance to the use of FINAL, rather than saying "avoid using it"

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Updated the SELECT...FROM...FINAL documentation to add better and more practical guidance to the use of FINAL, rather than saying "avoid using it"

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 6e0e4d154dba78e009ada751d7ecd753c2f1db61 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58137/6e0e4d154dba78e009ada751d7ecd753c2f1db61/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58137/6e0e4d154dba78e009ada751d7ecd753c2f1db61/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: docs/en/sql-reference/statements/select/from.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Queries that use `FINAL` are executed slightly slower than similar queries that
 - Data is merged during query execution.
 - Queries with `FINAL` read primary key columns in addition to the columns specified in the query.
 
-**In most cases, avoid using `FINAL`.** The common approach is to use different queries that assume the background processes of the `MergeTree` engine haven’t happened yet and deal with it by applying aggregation (for example, to discard duplicates).
+`FINAL` requires additional compute and memory resources, as the processing that normally would occur at merge time must occur in memory at the time of the query. However, using FINAL is sometimes necessary in order to produce accurate results, and is less expensive than running `OPTIMZIE` to force a merge. It is also sometimes possible to use different queries that assume the background processes of the `MergeTree` engine haven’t happened yet and deal with it by applying aggregation (for example, to discard duplicates). If you need to use FINAL in your queries in order to get the required results, then it is okay to do so but be aware of the additional processing required.
```
Comment (rfraposa) [inline docs/en/sql-reference/statements/select/from.md:37]: `OPTIMZIE` should be `OPTIMIZE`

**Review by UnamedRus — COMMENTED** [review]
> , and is less expensive than running `OPTIMIZE` to force a merge.

OPTIMIZE FINAL is not an option, not just because of expensiveness, but because you can't run it after each insert. (and you aren't going to wait until completion of OPTIMIZE for SELECT queries)

> Queries with `FINAL` read primary key columns in addition to the columns specified in the query.

Not always
https://github.com/ClickHouse/ClickHouse/issues/43296
https://github.com/ClickHouse/ClickHouse/pull/53919

> However, using FINAL is sometimes necessary 
> then it is okay to do so but be aware of the additional processing required.

If people need to get rid of duplicates, there are not a lot of options, and all of them need additional processing.
At this current stage FINAL provide pretty decent performance, and beat any other option almost for all possible use cases, plus it much simpler to use and can be combined with read in order/ aggregation in order (in future)

So, wording should be more confident, IMO.

Three more good FINAL improvement is on it's way -> 
https://github.com/ClickHouse/ClickHouse/pull/58120
https://github.com/ClickHouse/ClickHouse/pull/54366
https://github.com/ClickHouse/ClickHouse/pull/58142

**Review by alexey-milovidov — APPROVED** [review]
LGTM

**Pull request merged**